### PR TITLE
Remove remaining instances of "rate tick" from RFQ systems

### PR DESCRIPTION
**Pull request by ffranr**
This PR removes any remaining mention of "rate tick" from the Price Oracle RPC endpoints and the RFQ packages.

### CODE REVIEW

**Comment by ffranr** [comment]
The remaining RFQ cleanup work after this PR is merge:

* Add request message fields: `suggestedOutAssetRate` and `transferType`.
* Rearrange fields in request message to correspond with BLIP (refactor only, no logic changes, TLV values should be correct after current PR is merged).

**Review by dstadulis — CHANGES_REQUESTED** [review]

File: docs/examples/basic-price-oracle/main.go [hunk 1]
```diff
@@ -142,61 +141,61 @@ func getRateTick(transactionType oraclerpc.TransactionType,
...
 //
 // Example use case:
 //
 // Alice is trying to pay an invoice by spending an asset. Alice therefore
 // requests that Bob (her asset channel counterparty) purchase the asset from
 // her. Bob's payment, in BTC, will pay the invoice.
 //
-// Alice requests a bid quote from Bob. Her request includes a rate tick
-// hint (ask). Alice get the rate tick hint by calling this endpoint. She sets:
+// Alice requests a bid quote from Bob. Her request includes an asset rates hint
```
[the remark this answers is not in the document]
Comment (ffranr) [reply docs/examples/basic-price-oracle/main.go:172]: The request includes an in-asset to BTC rate hint and an out-asset to BTC rate hint.

File: docs/examples/basic-price-oracle/main.go [hunk 2]
```diff
@@ -142,61 +141,61 @@ func getRateTick(transactionType oraclerpc.TransactionType,
...
+// (ask). Alice get the asset rates hint by calling this endpoint. She sets:
```
[the remark this answers is not in the document]
Comment (ffranr) [reply docs/examples/basic-price-oracle/main.go:173]: I've gone with "Alice obtains". Let me know if you think "receives" is better or something else.

File: taprpc/priceoraclerpc/price_oracle.swagger.json [hunk 3]
```diff
@@ -161,6 +161,25 @@
...
+    "priceoraclerpcAssetRates": {
+      "type": "object",
+      "properties": {
+        "subjectAssetRate": {
+          "$ref": "#/definitions/priceoraclerpcFixedPoint",
+          "description": "subjectAssetRate is the number of subject asset units per BTC represented\nas a fixed-point number. This field is also commonly referred to as the\nsubject asset to BTC (conversion) rate.  When the subject asset is BTC\nmilli-satoshis (msats), this field should be set to 100 billion since\nthere are 100 billion msats in a BTC."
+        },
+        "paymentAssetRate": {
+          "$ref": "#/definitions/priceoraclerpcFixedPoint",
+          "description": "paymentAssetRate is the number of payment asset units per BTC represented\nas a fixed-point number. This field is also commonly referred to as the\npayment asset to BTC (conversion) rate. When the payment asset is BTC\nmilli-satoshis (msats), this field should be set to 100 billion since\nthere are 100 billion msats in a BTC."
```
[the remark this answers is not in the document]
Comment (ffranr) [reply taprpc/priceoraclerpc/price_oracle.swagger.json:173]: I've gone with:
```
When the payment asset is BTC,
    // this field should be set to 100 billion, as one BTC is equivalent to 100
    // billion msats.
```

File: taprpc/priceoraclerpc/price_oracle.swagger.json [hunk 4]
```diff
@@ -213,50 +232,31 @@
...
-          "$ref": "#/definitions/priceoraclerpcQueryRateTickSuccessResponse",
-          "description": "success is the successful response to the rate tick query."
-        },
-        "error": {
-          "$ref": "#/definitions/priceoraclerpcQueryRateTickErrResponse",
-          "description": "error is the error response to the rate tick query."
-        }
-      },
-      "description": "QueryRateTickResponse is the response to a rate tick query."
+      "description": "QueryAssetRatesErrResponse is the error response to a QueryAssetRates call."
```
[the remark this answers is not in the document]
Comment (ffranr) [reply taprpc/priceoraclerpc/price_oracle.swagger.json:235]: `QueryAssetRatesErrResponse` encapsulates all error responses from a price oracle service `QueryAssetRates` endpoint call.

File: taprpc/priceoraclerpc/price_oracle.pb.go [hunk 5]
```diff
@@ -590,49 +588,49 @@ func (x *QueryRateTickResponse) ProtoReflect() protoreflect.Message {
 	return mi.MessageOf(x)
 }
 
-// Deprecated: Use QueryRateTickResponse.ProtoReflect.Descriptor instead.
-func (*QueryRateTickResponse) Descriptor() ([]byte, []int) {
+// Deprecated: Use QueryAssetRatesResponse.ProtoReflect.Descriptor instead.
+func (*QueryAssetRatesResponse) Descriptor() ([]byte, []int) {
```
[the remark this answers is not in the document]
Comment (ffranr) [reply taprpc/priceoraclerpc/price_oracle.pb.go:592]: This was auto-generated by our protos generator. It's probably in place for backwards compatibility. I think we can ignore this.

File: docs/examples/basic-price-oracle/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dstadulis) [reply docs/examples/basic-price-oracle/main.go]: obtains sgtm

**Pull request merged**